### PR TITLE
Add Delta lake upsert blog

### PR DESCRIPTION
**Pull request by avriiil**
This PR adds a blog about "delta lake upsert"

### CODE REVIEW

File: src/blog/delta-lake-upsert/index.mdx [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+    )
+```
+
+Read more about using Delta Lake without Spark dependencies in the [Delta Lake without Spark post](https://delta.io/blog/delta-lake-without-spark/).
+
+## Best Practices for Delta Lake Upserts
+
+Here are some best practices to consider when running your Delta Lake upsert operations:
+
+- **Optimize Table Performance**: Periodically optimize and vacuum yourDelta tables to improve performance and reduce storage costs. Read more in the [Delta Lake Optimize post](https://delta.io/blog/delta-lake-optimize/).
```
Comment (newfront) [inline src/blog/delta-lake-upsert/index.mdx:356]: typo: there is a missing space between `yourDelta`, should be your Delta tables.

File: src/blog/delta-lake-upsert/index.mdx [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+```python
+> %%time
+> df_parq.write.format("parquet").save("tmp/new_data.parquet")
+
+CPU times: user 9.09 ms, sys: 5.65 ms, total: 14.7 ms
+Wall time: 28.8 s
+```
+
+This took more than 28seconds on an M1 Macbook Pro with 8 cores. That’s quite slow for a single upsert operation.
```
Comment (newfront) [inline src/blog/delta-lake-upsert/index.mdx:98]: typo: missing space between 28 and seconds.

**Review by newfront — CHANGES_REQUESTED** [review]
Just the two small typos and this is good to go. :)

**Comment by avriiil** [comment]
thanks @newfront, this should be good to go now!

**Review by newfront — APPROVED** [review]
This looks good to me. :shipit:

**Comment by newfront** [comment]
@dennyglee could you give this a +1 and help get this out the door?

**Pull request merged**